src/osd/scrubber/scrub_machine.cc

```diff
@@ -277,18 +277,25 @@ sc::result ReservingReplicas::react(const ReplicaReject& ev)
   DECLARE_LOCALS;  // 'scrbr' & 'pg_id' aliases
   auto& session = context<Session>();
   dout(10) << "ReservingReplicas::react(const ReplicaReject&)" << dendl;
-  session.m_reservations->log_failure_and_duration(scrbcnt_resrv_rejected);
 
-  // manipulate the 'next to reserve' iterator to exclude
-  // the rejecting replica from the set of replicas requiring release
-  session.m_reservations->verify_rejections_source(ev.m_op, ev.m_from);
+  // Verify that the message is from the replica we were expecting a reply from,
+  // and that the message is not stale. If all is well - this is a real rejection:
+  // - log required details;
+  // - manipulate the 'next to reserve' iterator to exclude
+  //   the rejecting replica from the set of replicas requiring release
+  if (session.m_reservations->handle_rejection(ev.m_op, ev.m_from)) {
```

Comment:
Seems like it would probably be easier for this to just be

```
if (session.m_reservations->handle_rejection(ev.m_op, ev.m_from)) {
     return discard_event();
}
```

Though, handle_rejection should probably be something like is_reservation_response_valid.